src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -23,5 +24,32 @@ public static class MemoryMarshal
         /// </remarks>
         public static Memory<T> AsMemory<T>(ReadOnlyMemory<T> readOnlyMemory) =>
             Unsafe.As<ReadOnlyMemory<T>, Memory<T>>(ref readOnlyMemory);
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ref T GetReference<T>(in Span<T> span) => ref span.Reference;
```

Comment:
Have you tried whether it is worth it to have `in` here? (It was discussed in the issue.)

Reply:
Using `in` doesn't really help (at least for fast span). And if we don't inline the method, it actually hurts performance.

If I pass a Span using `in`, it seems to hurt performance in some cases:

```C#
[MethodImpl(MethodImplOptions.NoInlining)]
public static void CallTestMethodWithIn(Span<int> span)
{
    int temp = TestMethodWithIn(span);
}

[MethodImpl(MethodImplOptions.NoInlining)]
public static void CallTestMethodWithoutIn(Span<int> span)
{
    int temp = TestMethodWithoutIn(span);
}

[MethodImpl(MethodImplOptions.NoInlining)]
public static int TestMethodWithIn(in ReadOnlySpan<int> span)
{
    int val = 0;
    for(int i = 0; i < span.Length; i++)
        val ^= span[i];
    return val;
}

[MethodImpl(MethodImplOptions.NoInlining)]
public static int TestMethodWithoutIn(ReadOnlySpan<int> span)
{
    int val = 0;
    for (int i = 0; i < span.Length; i++)
        val ^= span[i];
    return val;
}
```
**Runtime (s):**
CallTestMethodWithoutIn: 6.3821768
CallTestMethodWithIn: 6.8055258

There is an extra compare and jump in the loop body if I pass the Span with `in`.
![image](https://user-images.githubusercontent.com/6527137/33751106-67306588-db8d-11e7-8b2e-a022d36faf4f.png)